scripts/2.TCGA-MLexample-covariates.py

```diff
@@ -281,26 +284,33 @@ def get_threshold_metrics(y_true, y_pred):
 }
 
 
-# In[22]:
-
-def get_coefficients(classifier, X_mat):
-    coef_df = pd.DataFrame.from_items([
-        ('feature', X_mat.columns),
-        ('weight', classifier.coef_[0]),
-    ])
+# In[19]:
 
+def get_coefficients(classifier, feature_set):  
```

Comment:
@patrick-miller IMO it's a real sklearn deficiency / bug to not be able to extract labels and provenance info for each feature: see https://github.com/scikit-learn/scikit-learn/issues/7536.

We created a [workaround here](https://github.com/cognoma/cognoml/blob/f9669edfe0871cd161ffefa3409f67c98dd407a2/cognoml/utils.py#L62-L94). I'm not sure if it will work on your more advanced pipeline, but worth a try.

 > The only reason I see for not moving it is if the researchers wants to change the dimensionality reduction, the function as-is will break.

I don't think it's realistic for most users to be able to change this, since it requires quite intricate knowledge of sklearn. I say we aim for a robust workaround, but if not possible we can use some hacks. Also if we can detect the failure cases, we can just return feature names like `X1`, `X2`...